gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -69,6 +69,10 @@ protected Optional<Dag<JobExecutionPlan>> initialize(DagManagementStateStore dag
       Optional<Dag<JobExecutionPlan>> dag = this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec).toJavaUtil();
       if (dag.isPresent()) {
         dagManagementStateStore.checkpointDag(dag.get());
+        // Delete adhoc flowSpecs from catalog after checkpointing the dag to be worked on
+        if (!flowSpec.isScheduled()) {
+          dagManagementStateStore.removeFlowSpec(flowSpec);
```

Comment:
Is it the right place to remove flow spec? If the LaunchDagProc act fails, a second retry will not be able to compile.
should we do it in LaunchDagTask conclude ?

Reply:
good catch, moved it to the LaunchDagTask conclude